src/main/scala/rules/SymbolicExecutionRules.scala

```diff
@@ -32,9 +33,8 @@ trait SymbolicExecutionRules extends Immutable {
         val ce_type = Verifier.config.counterexample()
         var ce: Counterexample = ce_type match {
           case "native" => 
-            val oldHeap = if (s.oldHeaps.contains(Verifier.PRE_STATE_LABEL))
-              Some(s.oldHeaps(Verifier.PRE_STATE_LABEL).values)
-            else None
+            val optOldHeap: Option[Heap] = s.oldHeaps.get(Verifier.PRE_STATE_LABEL)

```

Comment:
Can't you replace lines 36 and 37 with `s.oldHeaps.get(Verifier.PRE_STATE_LABEL).map(_.values)`?

Do you use IntelliJ? It's pretty good at recommending such simplifications, at least for chained expressions.

Reply:
I am currently using VSCode but I guess I will check out IntelliJ then.
Thanks for the suggestion, it does work. I was not aware that map works like that on Option instances.